mqtt/mqtt-bridge/src/pump/egress.rs

```diff
@@ -59,26 +69,26 @@ where
         info!("starting egress publication processing...");
 
         let mut shutdown = shutdown_recv.fuse();
-        let mut loader = store.loader().fuse();
+
+        // Take the stream of loaded messages and convert to a stream of futures which publish.
+        // Then convert to buffered stream so that we can have multiple in-flight and also limit number of publications.
+        let loader = store.loader();
+        let publish_handle_stream = stream::iter(iter::repeat(publish_handle.clone()));
```

Comment:
Lmk if there is a better way. We can't move the publish handle into the map closure.

Reply:
I came up with 
```rust
let publications = store
            .loader()
            .filter_map(|loaded| {
                let publish_handle = publish_handle.clone();
                async {
                    match loaded {
                        Ok((key, publication)) => {
                            Some(try_publish(key, publication, publish_handle))
                        }
                        Err(e) => {
                            error!(err = %e, "failed loading publication from store");
                            None
                        }
                    }
                }
            })
            .buffer_unordered(MAX_IN_FLIGHT);
        pin_mut!(publications);
...

...
async fn try_publish(key: Key, publication: Publication, mut publish_handle: PublishHandle) -> Key {
    debug!("publishing {:?}", key);
    if let Err(e) = publish_handle.publish(publication).await {
        error!(err = %e, "failed publish");
    }

    key
}
```

Reply:
I'm guessing you didn't try to build this? I tried this Friday and it didn't work.

Reply:
actually I did and it compiled at least :)